Update Spatial Material 3D page for Godot 4 (#4587)
godotengine/godot-docs: tutorials/3d/standard_material_3d.rst

.. _doc_standard_material_3d:

Standard Material 3D and ORM Material 3D
========================================

Introduction
------------

> ``StandardMaterial3D`` are ``ORMMaterial3D`` default 3D materials that aim to

Reviewer: ```suggestion
``StandardMaterial3D`` and ``ORMMaterial3D`` (Occlusion, Roughness, Metallic) default 3D materials that aim to
```

are default 3D materials that aim to provide most of the features artists look
for in a material, without the need for writing shader code. However, they can
be converted to shader code if additional functionality is needed.

This tutorial explains the parameters present in both materials.

> There are three ways to add a these materials to an object. It can be added in

Reviewer 2: ```suggestion
There are three ways to add these materials to an object. It can be added in
```

> There are three ways to add a these materials to an object. It can be added in

Reviewer 2: ```suggestion
There are three ways to add a these materials to an object. A material can be added in
```

Making this change as the previous sentence now uses plural ("these materials"), while the following sentences still use singular ("it can be added"). Hope this change makes it a bit clearer :) We could also simply change "it" to plural "they" but that seems a bit confusing, maybe.

It will also override the regular material property of the node and the material property of
the mesh.

BaseMaterial 3D Settings
------------------------

StandardMaterial3D has many settings that determine the look of a material. All of these are
under the BaseMaterial3D category

.. image:: img/spatial_material1.png

ORM materials are almost exactly the same with one difference. Instead of separate settings

> and textures for roughness and metallic, there is a single ORM texture.

Reviewer: ```suggestion
and textures for occlusion,roughness, and metallic, there is a single ORM texture.
```

> and textures for occlusion, roughness, and metallic, there is a single ORM texture.

Reviewer 2: ```suggestion
and textures for occlusion, roughness, and metallic, there is a single ORM texture.
```

Maybe add a single sentence how ORM textures work (i.e. containing one map in each color channel, I assume?), and how they're commonly authored? May make sense for newer users to mention this briefly for context.

For this reason, materials in Godot are assumed opaque unless
specified otherwise. The main settings that enable transparency are:

* Transparency (this one)
* Blend mode set to other than "Mix"
* Enabling distance or proximity fade

When transparency other than ``0`` or ``1`` is not needed, it's possible to
set a threshold to prevent the object from rendering semi-transparent pixels

> using the alpha scissor opetion.

Reviewer: option typo

.. image:: img/spatial_material12.png

> This renders the object via the opaque pipeline, which is faster and allows it

Reviewer: The object only goes through the opaque pipeline when opaque pre-pass is on

to use mid- and post-process effects such as SSAO, SSR, etc.

Blend Mode
~~~~~~~~~~

Controls the blend mode for the material. Keep in mind that any mode
other than *Mix* forces the object to go through the transparent pipeline.

* **Mix:** Default blend mode, alpha controls how much the object is visible.

> * **Add:** Object is blended additively, nice for flares or some

Reviewer: Maybe "the final color of the object is added to the color of the screen..."

And a similar phrase for the below two

intermediate values generally works best.

Clearcoat
---------

> *This feature is only available when using the Vulkan backend.*

Reviewer: I would remove these warnings for now. We can re add some if they turn out to be relevant

.. image:: img/spatial_material20.png

Subsurface Scattering
---------------------

*This feature is only available when using the Vulkan backend.*

This effect emulates light that penetrates an object's surface, is scattered,
and then comes out. It is useful to create realistic skin, marble, colored
liquids, etc.

.. image:: img/spatial_material21.png

Back Lighting
-------------

> Makes the material look like it's being lit from behind.

Reviewer: The description for Transmission is still relevant here

.. image:: img/spatial_material25.png

World Triplanar
~~~~~~~~~~~~~~~

When using triplanar mapping triplanar is computed in object local space. This
option makes triplanar work in world space.

Sampling
--------

Filter
~~~~~~

Shadows
-------

> 

Reviewer: It looks like these still need to be filled in

.. image:: img/spatial_material25.png

World Triplanar
~~~~~~~~~~~~~~~

> When using triplanar mapping triplanar is computed in object local space. This
> option makes triplanar work in world space.

Reviewer 2: ```suggestion
When using triplanar mapping, it is computed in object local space. This
option makes it use world space instead.
```
